test/language.c

```diff
@@ -57,47 +57,55 @@ int main(void)
-    TEST_LANG_GUESS("foo.gsw-u-sd-chzh.srt", "gsw-u-sd-chzh", 3, false);
-    TEST_LANG_GUESS("foo.en-.srt", "", -1, false);
-    TEST_LANG_GUESS("foo.en-US-.srt", "", -1, false);
-    TEST_LANG_GUESS("foo.en-aaaaaaaaa.srt", "", -1, false);
-    TEST_LANG_GUESS("foo.en-0.srt", "", -1, false);
+    TEST_LANG_GUESS("foo.en-US.srt", "en-US", 3, 0);
+    TEST_LANG_GUESS("foo.en-US.hi.srt", "en-US", 3, TRACK_HEARING_IMPAIRED);
+    TEST_LANG_GUESS("foo.en-US.sdh.srt", "en-US", 3, TRACK_HEARING_IMPAIRED);
+    TEST_LANG_GUESS("foo.en-US.forced.srt", "en-US", 3, TRACK_FORCED);
+    TEST_LANG_GUESS("foo.en-US.forced.sdh.srt", "en-US", 3, TRACK_HEARING_IMPAIRED | TRACK_FORCED);
+    TEST_LANG_GUESS("foo.en-US.sdh.forced.srt", "en-US", 3, TRACK_HEARING_IMPAIRED | TRACK_FORCED);
+    TEST_LANG_GUESS("foo.en-simple.srt", "en-simple", 3, 0);
+    TEST_LANG_GUESS("foo.sgn-FSL.srt", "sgn-FSL", 3, 0);
+    TEST_LANG_GUESS("foo.gsw-u-sd-chzh.srt", "gsw-u-sd-chzh", 3, 0);
+    TEST_LANG_GUESS("foo.en-.srt", "", -1, 0);
+    TEST_LANG_GUESS("foo.en-US-.srt", "", -1, 0);
+    TEST_LANG_GUESS("foo.en-aaaaaaaaa.srt", "", -1, 0);
+    TEST_LANG_GUESS("foo.en-0.srt", "", -1, 0);
 
-    TEST_LANG_GUESS("foo[en].srt", "en", 3, false);
-    TEST_LANG_GUESS("foo[en-US].srt", "en-US", 3, false);
-    TEST_LANG_GUESS("foo[en-US][hi].srt", "en-US", 3, true);
-    TEST_LANG_GUESS("foo[en-US][sdh].srt", "en-US", 3, true);
-    TEST_LANG_GUESS("foo[].srt", "", -1, false);
+    TEST_LANG_GUESS("foo[en].srt", "en", 3, 00);
+    TEST_LANG_GUESS("foo[en-US].srt", "en-US", 3, 0);
+    TEST_LANG_GUESS("foo[en-US][hi].srt", "en-US", 3, TRACK_HEARING_IMPAIRED);
+    TEST_LANG_GUESS("foo[en-US][sdh].srt", "en-US", 3, TRACK_HEARING_IMPAIRED);
+    TEST_LANG_GUESS("foo[en-US][forced].srt", "en-US", 3, TRACK_FORCED);
+    TEST_LANG_GUESS("foo[en-US][forced][sdh].srt", "en-US", 3, TRACK_HEARING_IMPAIRED | TRACK_FORCED);
+    TEST_LANG_GUESS("foo[en-US][sdh][forced].srt", "en-US", 3, TRACK_HEARING_IMPAIRED | TRACK_FORCED);
+    TEST_LANG_GUESS("foo[].srt", "", -1, 0);
 
-    TEST_LANG_GUESS("foo(en).srt", "en", 3, false);
-    TEST_LANG_GUESS("foo(en-US).srt", "en-US", 3, false);
-    TEST_LANG_GUESS("foo(en-US)(hi).srt", "en-US", 3, true);
-    TEST_LANG_GUESS("foo(en-US)(sdh).srt", "en-US", 3, true);
-    TEST_LANG_GUESS("foo().srt", "", -1, false);
+    TEST_LANG_GUESS("foo(en).srt", "en", 3, 0);
+    TEST_LANG_GUESS("foo(en-US).srt", "en-US", 3, 0);
+    TEST_LANG_GUESS("foo(en-US)(hi).srt", "en-US", 3, TRACK_HEARING_IMPAIRED);
+    TEST_LANG_GUESS("foo(en-US)(sdh).srt", "en-US", 3, TRACK_HEARING_IMPAIRED);
+    TEST_LANG_GUESS("foo(en-US)(forced).srt", "en-US", 3, TRACK_FORCED);
+    TEST_LANG_GUESS("foo(en-US)(forced)(sdh).srt", "en-US", 3, TRACK_HEARING_IMPAIRED | TRACK_FORCED);
+    TEST_LANG_GUESS("foo(en-US)(sdh)(forced).srt", "en-US", 3, TRACK_HEARING_IMPAIRED | TRACK_FORCED);
+    TEST_LANG_GUESS("foo().srt", "", -1, 0);
```

Comment:
Add also some tests for tags without language. For example
```
    TEST_LANG_GUESS("foo.hi.forced.srt", "", -1, TRACK_HEARING_IMPAIRED | TRACK_FORCED);
    TEST_LANG_GUESS("foo.forced.hi.srt", "", -1, TRACK_HEARING_IMPAIRED | TRACK_FORCED);
    TEST_LANG_GUESS("foo.hi.srt", "", -1, TRACK_HEARING_IMPAIRED);
    TEST_LANG_GUESS("foo.forced.srt", "", -1, TRACK_FORCED);
```

Not sure how useful that is, but it works like that currently.